Revamp Developer Portal: Update: Document Custom SSO Login URL (v1.16.0) (#1928)
TykTechnologies/tyk-docs: tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso.mdx

> <img src="/img/dashboard/portal-management/enterprise-portal/portal-sso-login.png" alt="SSO Profile Details" />
> 
> Tyk Enterprise Developer Portal doesn't supply a login page for Single Sign-On out of the box, so you might need to create one.
> From Portal v1.16.0, you can alternatively configure a [custom SSO login URL](#custom-sso-login-url) to redirect users directly to your external IdP login page without building one.
> Here is an example of such a page that works with a profile for the LDAP identity management system:

Author: ```suggestion
Tyk Enterprise Developer Portal doesn't supply a login page for Single Sign-On out of the box, so you might need to create one.

<Note>
From Portal v1.16.0, you can alternatively configure a [custom SSO login URL](#custom-sso-login-url) to redirect users directly to your external IdP login page without building one.
<Note/>

Here is an example of such a page that works with a profile for the LDAP identity management system:
```

> Any changes to the TIB profile will be effective after restarting your TIB instance.
> 
> 2. Create a login page for admin users. We don't supply a login page for Single Sign-On out of the box, so you need to create one.
>    From Portal v1.16.0, you can alternatively configure a [custom SSO login URL](#custom-sso-login-url) to redirect users directly to your external IdP login page without building one.
> Here is an example of such page that works with a profile for LDAP identity management system:

Author: ```suggestion
2. Create a login page for admin users. We don't supply a login page for Single Sign-On out of the box, so you need to create one.

   <Note>
   From Portal v1.16.0, you can alternatively configure a [custom SSO login URL](#custom-sso-login-url) to redirect users directly to your external IdP login page without building one.
  <Note/>

Here is an example of such page that works with a profile for LDAP identity management system:
```

2. Create a login page for developers. We don't supply a login page for Single Sign-On out of the box, so you need to create one.

>    From Portal v1.16.0, you can alternatively configure a [custom SSO login URL](#custom-sso-login-url) to redirect users directly to your external IdP login page without building one.

Author: ```suggestion
```

```
3. Now you should be able to log in to the portal with your identity provider as a developer

## Custom SSO Login URL

> *Available from Portal v1.16.0. See the [v1.16.0 release notes](/developer-support/release-notes/portal#1-16-0-release-notes).*

Author: ```suggestion
*Available from Portal v1.16.0. See the [v1.16.0 release notes](/developer-support/release-notes/portal#custom-sso-login-url-configuration).*
```

By default, the Developer Portal renders its own login page at `/auth/password/login`. When you set `SSOCustomLoginURL`, the portal automatically redirects all login requests to the URL you specify (typically your identity provider's login page) instead of displaying the built-in form. The portal's **Log in** button also points to this URL.

Use this setting when you want to delegate authentication entirely to an external IdP without exposing the portal's default login page to users.

<Note>
This setting only redirects the login entry point. User registration and password reset pages are not affected.
</Note>

### Configuration

Set either the environment variable or the equivalent configuration file key:

<Tabs>
<Tab title="Environment variable">
```ini
PORTAL_SSO_CUSTOM_LOGIN_URL=https://your-idp.example.com/login
```
</Tab>
<Tab title="Config file">
```json
{
  "SSOCustomLoginURL": "https://your-idp.example.com/login"
}
```
</Tab>
</Tabs>

For the full reference, see [`PORTAL_SSO_CUSTOM_LOGIN_URL`](/product-stack/tyk-enterprise-developer-portal/deploy/configuration#portal_sso_custom_login_url) in the configuration reference.

> For a complete end-to-end walkthrough of setting up IDP-based SSO, from creating a TIB profile to testing the full login flow, see the End-to-End SSO Setup guide in the How-To Guides section.

Author: ```suggestion
```

```
3. Now you should be able to log in to the portal with your identity provider as a developer

## Custom SSO Login URL

*Available from Portal v1.16.0. See the [v1.16.0 release notes](/developer-support/release-notes/portal#1-16-0-release-notes).*

> By default, the Developer Portal renders its own login page at `/auth/password/login`. When you set `SSOCustomLoginURL`, the portal automatically redirects all login requests to the URL you specify (typically your identity provider's login page) instead of displaying the built-in form. The portal's **Log in** button also points to this URL.

Reviewer: ```suggestion
By default, the Developer Portal renders its own login page at `/auth/password/login`. When you set `SSOCustomLoginURL`, the portal automatically redirects all login requests to the URL you specify (typically your identity provider's login page) instead of displaying the built-in form. The Portal's **Log in** button also points to this URL.
```

> Use this setting when you want to delegate authentication entirely to an external IdP without exposing the portal's default login page to users.

Reviewer: ```suggestion
Use this setting when you want to delegate authentication entirely to an external IdP without exposing the Portal's default login page to users.
```
